src/main/java/de/tum/in/www1/artemis/repository/hestia/CoverageReportRepository.java

```diff
@@ -31,31 +43,56 @@ public interface CoverageReportRepository extends ArtemisJpaRepository<CoverageR
     void deleteBySubmissionId(Long submissionId);
 
     @Query("""
-            SELECT DISTINCT r
+            SELECT new de.tum.in.www1.artemis.repository.hestia.CoverageReportAndSubmissionDate(r, s.submissionDate)
```

Comment:
Again, why select the report and its submission date only to later just use the id